nanoFirmwareFlasher.Library/FirmwareArchiveManager.cs

```diff
@@ -0,0 +1,221 @@
+                }
+            }
+            if (remoteTargets.Count == 0)
+            {
+                return ExitCodes.E9005;
+            }
+
+            ExitCodes result = ExitCodes.OK;
+
+            foreach ((CloudSmithPackageDetail remoteTarget, Version _) in remoteTargets.Values)
+            {
+                // Download the firmware
+                var package = new ArchiveFirmwarePackage(remoteTarget.Name, remoteTarget.Version, preview)
+                {
+                    Verbosity = verbosity
+                };
+                (ExitCodes exitCode, string fwFilePath) = await package.DownloadPackageAsync(_archivePath, null, false, false);
+                if (exitCode != ExitCodes.OK)
+                {
+                    result = exitCode;
+                    if (verbosity >= VerbosityLevel.Normal)
+                    {
+                        OutputWriter.ForegroundColor = ConsoleColor.Red;
+                        OutputWriter.WriteLine($"Could not download target {remoteTarget.Name} {remoteTarget.Version}");
+                        OutputWriter.ForegroundColor = ConsoleColor.White;
+                    }
+                    continue;
+                }
+                else if (verbosity > VerbosityLevel.Normal)
+                {
+                    OutputWriter.WriteLine($"Added target {remoteTarget.Name} {remoteTarget.Version} to the archive");
+                }
+
+                // Write the file with package information
+                var packageInformation = new PersistedPackageInformation
+                {
+                    IsPreview = preview,
+                    Name = remoteTarget.Name,
+                    Platform = platform.ToString(),
+                    Version = remoteTarget.Version,
+                };
+                File.WriteAllText(
+                    $"{(fwFilePath.EndsWith(".zip") ? fwFilePath : Path.GetDirectoryName(fwFilePath))}{INFOFILE_EXTENSION}",
+                    JsonConvert.SerializeObject(packageInformation)
+                );
+            }
+            return result;
+        }
+        #endregion
+
+        #region Auxiliary classes that should only be used by the archive manager
+
+        #region Persisted firmware information
+        /// <summary>
+        /// Class with details of a CloudSmith package. The base class is the one
+        /// used for listing targets. The extra fields are required to select the
+        /// correct package for the firmware that should be installed.
+        /// </summary>
+        private sealed class PersistedPackageInformation : CloudSmithPackageDetail
+        {
+            /// <summary>
+            /// Platform code
+            /// </summary>
+            public string Platform { get; set; }
+
+            /// <summary>
+            /// Indicates whether this is a preview
+            /// </summary>
+            public bool IsPreview { get; set; }
+        }
+        private const string INFOFILE_EXTENSION = ".json";
```

Comment:
Empty line before and after declaration please.
Why isn't this private const at the top of the class as the code style calls for?

Reply:
Forgot to put it there. In my code I never put private fields at the top. They usually start to appear in other places in the code when they are needed, and I have to remember to move them to the top.